src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -429,8 +429,8 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
 
         return cephadm_config, []
 
-    def post_remove(self, daemon: DaemonDescription, is_failed_deploy: bool) -> None:
-        super().post_remove(daemon, is_failed_deploy=is_failed_deploy)
+    def post_remove(self, daemon: DaemonDescription, keep_keyring: bool) -> None:
+        super().post_remove(daemon, keep_keyring=keep_keyring)
         self.remove_keyring(daemon)
```

Comment:
I think I'm missing something with how this keep_keyring param works. It looks like we pass it up to the post_remove for the CephadmService class post_remove (since that's the class we're inheriting from) which doesn't make any use of the parameter and then after that runs we run remove_keyring regardless of what keep_keyring is set to.

Reply:
great catch